### PR TITLE
Adding Fs TC_CEPH-83586730

### PR DESCRIPTION
**Pull request by AmarnatReddy**
# Description
Adding Fs TC_CEPH-83586730

Steps:
        1. Set max_mds 1
        2. allow standby reply 
        3. check the gid of the standby reply
        4. set mds_inject_health_dummy on stanby reply node
        5. check the gid again. expectation is  There should not be any change in gid of standby-reply node

Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EGMDBZ/

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        stanby_reply_before, rc = client1.exec_command(
+            sudo=True, cmd="ceph fs dump -f json"
+        )
+        info_before = get_info(json.loads(stanby_reply_before), fs_name)
+        standby_replay_dict_before = {
+            key: value
+            for key, value in info_before.items()
+            if "standby-replay" in value["state"]
+        }
+        import pdb
```
Comment (hkadam134) [inline tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py:66]: can you remove pdb ?

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        )
+        client1.exec_command(
+            sudo=True,
+            cmd=f"cd {fuse_mount_dir};"
+            f"mkdir subdir;"
+            f"dd if=/dev/urandom of=subdir/sixmegs bs=1M conv=fdatasync count=6 seek=0;",
+        )
+        time.sleep(60)
+
+        sstanby_reply_after, rc = client1.exec_command(
```
Comment (hkadam134) [inline tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py:101]: sstanby --> typo ?

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            sudo=True, cmd=f"ceph fs set {fs_name} allow_standby_replay 0"
+        )
+        client1.exec_command(
+            sudo=True, cmd="ceph config set mon mon_allow_pool_delete true"
+        )
+        fs_util.remove_fs(client1, fs_name)
+
+
+@retry(CommandFailed, tries=3, delay=30)
+def wait_for_standby_reply_mds(fs_util, client1, fs_name):
```
Comment (hkadam134) [inline tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py:145]: can this be moved to utils file ?

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    standby_reply_mds = fs_util.get_standby_replay_mdss(client1, fs_name=fs_name)
+    if not standby_reply_mds:
+        raise CommandFailed("No Stanby Reply MDS")
+
+
+def get_info(json_output, fs_name):
```
Comment (hkadam134) [inline tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py:151]: rename to "get_filesystem_info" ?

Can this be moved to utils file ?

File: suites/quincy/cephfs/tier-3_cephfs_bugs.yaml [hunk 5]
```diff
@@ -222,3 +225,9 @@ tests:
       polarion-id: CEPH-83584084
       desc: Validate if no repeated mclientscaps(revoke) entries found in MDS.
       abort-on-fail: false
+  - test:
+      name: validate standby reply node not removed after setting mds_inject_health_dummy
```
Comment (neha-gangadhar) [inline suites/quincy/cephfs/tier-3_cephfs_bugs.yaml:229]: Please fix minor typo from "standby reply" to "standby replay"

**Comment by neha-gangadhar** [comment]
Please address the comments and looks good overall.

**Review by neha-gangadhar — APPROVED** [review]
lgtm

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py (hunk 1 above)
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py]: done

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py (hunk 4 above)
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py:140]: this is intended for this TC. so I kept here only

File: tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py (hunk 2 above)
Comment (AmarnatReddy) [reply tests/cephfs/cephfs_bugs/test_mds_inject_health_dummy.py]: done

**Review by sumabai — APPROVED** [review]
lgtm

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Review by hkadam134 — APPROVED** [review]
Looks Good

**Review by neha-gangadhar — DISMISSED** [review]
Approved

**Review by neha-gangadhar — APPROVED** [review]
lgtm

**Pull request merged**